src/grpc/lightning.ts

```diff
@@ -364,10 +365,35 @@ export function keysend(
             } else {
               if (state === 'IN_FLIGHT') {
                 // do nothing
-              } else if (state === 'FAILED_NO_ROUTE') {
-                reject(payment.failure_reason || payment)
-              } else if (state === 'FAILED') {
-                reject(payment.failure_reason || payment)
+              } else if (state === 'FAILED_NO_ROUTE' || state === 'FAILED') {
+                delete options.payment_hash
+                delete options.dest_custom_records[`${LND_KEYSEND_KEY}`]
+                delete options.dest_features
+                options.amp = true
+                console.log('Keysend failed, trying AMP')
```

Comment:
We might wanna try doing an amp payment first. Since if both users are using amp a keysend is more likely to fail so we might aswell use the option that will more likely succeed and more users will probably default to using amp on their nodes in the future

Reply:
yeah good point, and on lightning.engineering i read amp can do more and is a replacement for keysend. it can also be added to the docs like is done with keysend (telling users they need accept-amp=true)